docs/guides/migrating-cra.md

```diff
@@ -0,0 +1,774 @@
+
+Let's move our app's stylesheet and a few other assets to the `links` function in our root route:
+
+```jsx filename=root.jsx lines=[2,5,7-16,32]
+import * as React from "react";
+import { Links } from "@remix-run/react";
+
+import App from "./app";
+import stylesheetUrl from "./styles.css";
+
+export function links() {
+  // `links` returns an array of objects whose
+  // properties map to the `<link />` component props
+  return [
+    { rel: "icon", href: "/favicon.ico" },
+    { rel: "apple-touch-icon", href: "/logo192.png" },
+    { rel: "manifest", href: "/manifest.json" },
+    { rel: "stylesheet", href: stylesheetUrl },
+  ];
+}
+
+export default function Root() {
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta
+          name="viewport"
+          content="width=device-width, initial-scale=1"
+        />
+        <meta name="theme-color" content="#000000" />
+        <meta
+          name="description"
+          content="Web site created using create-react-app"
+        />
+        <Links />
+        <title>React App</title>
+      </head>
+      <body>
+        <App />
+      </body>
+    </html>
+  );
+}
+```
+
+You'll notice on line 32 that we've rendered a `<Links />` component that replaced all of our individual `<link />` components. This is inconsequential if we only ever use links in the root route, but it's all child routes may export their own links that will also be rendered here. The `links` function can also return a [`PageLinkDescriptor` object](../api/conventions#pagelinkdescriptor) that allows you to prefetch the resources for a page the user is likely to navigate to.
+
+If you inject `<link />` tags into your page client-side in your existing route components, either directly or via an abstraction like [`react-helmet`](https://www.npmjs.com/package/react-helmet), you'll get to delete a lot of code and possibly a dependency or two!
+
+### Route `<meta />` tags
+
+Similar to `links`, each route can also export a `meta` function that—you guessed it—returns a value responsible for rendering `<meta />` tags for that route. This is useful because each route often has its own
+
+The API is slightly different for `meta`. Instead of an array, it returns an object where the keys represent the meta `name` attribute (or `property` in the case of OpenGraph tags) and the value is the `content` attribute. The object can also accept a `title` property that renders a `<title />` component specifically for that route.
+
+Again—no more weird dances to get meta into your routes from deep in the component tree. Export them at the route level and let the server handle it. ✨
+
+## Using the `.jsx` extension for React component files
+
+In CRA you can use JSX in either `.jsx` or `.js` files. You'll need to rename any components with JSX to `.jsx`.
```

Comment:
```suggestion
In CRA you can use JSX in either `.jsx` or `.js` files. In Remix, you *must* use `.jsx` for all files that use JSX.
```

Reply:
Going to hold on this. @mjackson suggested that we should probably support JSX in `.js` files since esbuilt supports that directly now. May try to land that PR first!

Reply:
Related PR is up: https://github.com/remix-run/remix/pull/3059